src/workerinfo.js

```diff
@@ -0,0 +1,41 @@
+let taskcluster = require('taskcluster-client');
+let assert      = require('assert');
+let debug       = require('debug')('workerinfo');
+
+class WorkerInfo {
+  constructor(options) {
+    assert(options);
+    assert(options.Provisioner);
+    this.Provisioner = options.Provisioner;
+  }
+
+  async provisionerSeen(provisionerId) {
+    let expires = taskcluster.fromNow('5 days');  // temporary hard coded expiration
+
+    let provisioner = await this.Provisioner.load({provisionerId}, true);
+    if (provisioner) {
+      await provisioner.modify(entity => {
+        entity.expires = expires;
```

Comment:
This is very expensive...
You'll want to do:
```js
await provisioner.modify(provisioner => {
  // Update expiration, if we want to move it by more than 12 hours
  if (new Date(expires) - new Date(provisioner.expires) > 12 * 60 * 60 * 1000) {
    provisioner.expires = expires;
  }
})
```

Now it won't have to update more than 12 hours, sure expiration might be off by like, 12 hours,  but given that we want it to be something like 5 days anyways that doesn't really matter...